NIFI-11858 Configurable Column Name Normalization in PutDatabaseRecord and UpdateDatabaseTable

ravinarayansingh opened the pull request:
old PR link : https://github.com/apache/nifi/pull/7544 , https://github.com/apache/nifi/pull/8995

<!-- Licensed to the Apache Software Foundation (ASF) under one or more -->
<!-- contributor license agreements.  See the NOTICE file distributed with -->
<!-- this work for additional information regarding copyright ownership. -->
<!-- The ASF licenses this file to You under the Apache License, Version 2.0 -->
<!-- (the "License"); you may not use this file except in compliance with -->
<!-- the License.  You may obtain a copy of the License at -->
<!--     http://www.apache.org/licenses/LICENSE-2.0 -->
<!-- Unless required by applicable law or agreed to in writing, software -->
<!-- distributed under the License is distributed on an "AS IS" BASIS, -->
<!-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. -->
<!-- See the License for the specific language governing permissions and -->
<!-- limitations under the License. -->

# Summary

[NIFI-11858 ](https://issues.apache.org/jira/browse/NIFI-11858 )

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

mattyb149 commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java:
```diff
@@ -277,6 +285,24 @@ public class PutDatabaseRecord extends AbstractProcessor {
+    public static final PropertyDescriptor TRANSLATION_STRATEGY = new PropertyDescriptor.Builder()
+            .required(true)
+            .name("Column Name Translation Strategy")
+            .description("The strategy used to normalize table column name")
```
Do all the non-regex options uppercase the name for translation purposes? If so that's fine but somewhere (maybe here) the doc should reflect that no matter the strategy (unless possibly the new Regex one), the name will be uppercased to do case-insensitive matching

ravinarayansingh replied:
Hi @mattyb149 
I have updated the documentation

mattyb149 commented on nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/TranslationStrategy.java:
```diff
@@ -0,0 +1,57 @@
+            "Spaces will be removed from column names Ex. 'User Name' becomes 'USERNAME'"),
+    REMOVE_UNDERSCORE_AND_SPACE("Remove Underscores and Spaces",
+            "Spaces and Underscores will be removed from column names Ex. 'User_1 Name' becomes 'USER1NAME'"),
+    REMOVE_ALL_SPECIAL_CHAR("Remove Regular Expression Characters",
```
Can you include examples for the two Regex choices? They are the most complex so the more and clearer the documentation the better. After that I'm good to merge!

ravinarayansingh replied:
hi @mattyb149 
I have added example . please have a look

mattyb149 commented:
I think the way you're merging main into your feature branch (instead of rebasing) has often been causing issues. I will review when your PR branch can be safely rebased.

ravinarayansingh commented:
> I think the way you're merging main into your feature branch (instead of rebasing) has often been causing issues. I will review when your PR branch can be safely rebased.

Hi @mattyb149 
i have rebased this is following is the message  show rebase is successful
```
dropping 2e017bd2e9f3d7ec8673c60dd03a1b9c1e11e893 updated the description to reflect uppercase conversion of column name  uppercased to do case-insensitive matching irrespective of strategy -- patch contents already upstream
Successfully rebased and updated refs/heads/NIFI-11858-clean.
```
let me know what else i need to  do

mattyb149 commented:
+1 LGTM, tried various settings on both processors. Thanks for sticking with this! One note though, this latest PR doesn't have a commit with the Jira title in it, I added it when I squashed your 3 commits together and signed off on the single commit.
Thanks for the improvement! Merging to main
